nxt_editor/user_dir.py

```diff
@@ -48,7 +50,51 @@ def ensure_pref_dir_exists():
         raise Exception('Failed to generate user dir {}' + USER_DIR)
 
 
+def check_for_upgradable_prefs():
+    """
+    Identify preference files that can be safely upgraded
+    between major editor versions. Only existing preference files
+    from the nearest older version are copied; missing files are skipped
+    without warnings.
+    """
+    global UPGRADABLE_PREFS
+    global UPGRADE_PREFS_FROM_VERSION
+    for pref_file in JSON_PREFS:
+        if os.path.isfile(pref_file):
+            break
+    else:  # Didn't find any json prefs in current version prefs
+        dir_num = PREF_DIR_INT - 1
+        while dir_num > -1:
+            old_pref_dir = os.path.join(USER_DIR, PREF_DIR_NAME, str(dir_num))
+            for pref_file in JSON_PREFS:
+                file_name = os.path.basename(pref_file)
+                old_pref_file = os.path.join(old_pref_dir, file_name)
+                if os.path.isfile(old_pref_file):
+                    # In the future if we change the structure of the json
+                    # prefs we'll need a way to convert them or skip
+                    UPGRADABLE_PREFS.append(old_pref_file)
+            if UPGRADABLE_PREFS:
+                UPGRADE_PREFS_FROM_VERSION = dir_num
+                break
+            dir_num -= 1
+
+
+def upgrade_prefs():
+    """
+    Copies old 'upgradeable' prefs to current pref dir, will eat and
+    exception raised by shutil.copy. In the future this function may do more
+    than simply copy.
+    """
+    for pref_file in UPGRADABLE_PREFS:
```

Comment:
Probably you expected this question. Can we get away without the global here? Thinking if `check_for_upgradeable_prefs` returned a list that the dialog would pass to `upgrade_prefs(prefs_to_update)`?

Reply:
new comment clarifies, thanks